src/plugins/plugins_ex.h

```diff
@@ -386,119 +333,127 @@ struct breakpoint_by_pid_searcher
     }
 };
 
+template<typename Plugin, typename Params = Plugin>
+struct plugin_data
+{
+    Plugin* plugin;
+    Params* params;
+
+    plugin_data(Plugin* plugin, Params* params)
+        : plugin(plugin), params(params)
+    {};
+};
+
 class pluginex : public plugin
 {
 public:
     typedef event_response_t(*hook_cb_t)(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
 
-public:
     pluginex(drakvuf_t drakvuf, output_format_t output);
-    virtual ~pluginex();
+    virtual ~pluginex() {};
 
-public:
-    template<typename P = pluginex, typename Params = plugin_params<P>, typename IB,
-             typename AP = allocator<Params>, typename AT = allocator<drakvuf_trap_t>>
+    using trap_deleter = std::function<void(drakvuf_trap_t*)>;
+
+    // Params property is optional
+    template<typename Plugin = pluginex, typename Params = Plugin, typename IB>
```

Comment:
Actually, you don't have to pass a template argument here, just function argument will work :)
But yes, it would be nice to specify more clearly what types of functions can be passed here.